moto/cloudformation/responses.py

```diff
@@ -168,9 +193,15 @@ def create_change_set(self) -> str:
             for item in self._get_list_prefix("Tags.member")
         )
         parameters = {
-            param["parameter_key"]: param["parameter_value"]
+            param["parameter_key"]: stack.parameters[param["parameter_key"]]
+            if param.get("use_previous_value", "").lower() == "true"
+            and use_previous_template
```

Comment:
I've extracted this variable outside of the if-statement above so that it's always available - not just when updating a changeset. Otherewise there is a situation where Python would throw a `variable not declared` error.